zad1/main.rb

```diff
@@ -1,13 +1,25 @@
 def sort_letters(string)
-
+	string.chars.sort.join
 end
 
 def count_vowels(string)
-
+	result = 0
+	string.each_char do |x|
+		if['a', 'e', 'i', 'o', 'u', 'y'].include? x
+			result += 1
+		end
+	end
+	result
 end
 
 def filter_odd(array)
-
+	result = []
+	array.each do |x|
```

Comment:
Możemy się pozbyć zmiennej result przez wykorzystanie metody inject z prezentacji. Inject sam nam tworzy obiekt, który zwróci na końcu i wtedy `result` nam odpada. Można to oczywiście zrobić jeszcze ładniej, Array w Ruby posiada masę metod do właśnie takiego filtrowania np. [polecany select](https://ruby-doc.org/core-2.2.0/Array.html#method-i-select), ale Array jest też enumerablem więc ma posiada dużo metod do operacji na kolekcjach które można znaleźć [tutaj](https://ruby-doc.org/core-2.5.1/Enumerable.html)